lib/makara/proxy.rb

```diff
@@ -118,20 +124,20 @@ def strategy_class_for(strategy_name)
 
     def method_missing(m, *args, &block)
       if METHOD_MISSING_SKIP.include?(m)
-        return super(m, *args, &block)
+        return super
       end
 
       any_connection do |con|
-        if con.respond_to?(m)
-          con.public_send(m, *args, &block)
-        elsif con.respond_to?(m, true)
-          con.__send__(m, *args, &block)
+        if con.respond_to?(m, true)
+          con.send(m, *args, &block)
```

Comment:
Subtle regression here. We'll now call a private `#foo` method on the ConnectionWrapper instead of calling the public `#foo` method on the wrapped ConnectionAdapter. That's because using `#send` will dispatch the private method immediately whereas `#public_send` will fall back to method_missing, which is where the Makara ConnectionWrapper talks to the AR ConnectionWrapper.

Example
```ruby
c.send(:select, "select 1")
*** TypeError Exception: wrong argument type String (expected Array)

c.public_send(:select, "select 1")
   (2.1ms)  select 1
#<ActiveRecord::Result:0x00007f8784b4e7f0 @columns=["1"], @rows=[[1]], @hash_rows=nil, @column_types={}>
```

This is the cause of #326

Reply:
Thanks for catching this. TIL about `#send` not hitting `#method_missing`.

If you can send a PR and maybe add a spec to prevent a regression I'll get it merged ASAP.